assignments/this.js

```diff
@@ -11,16 +11,48 @@
 
 // Principle 1
 
-// code example for Window Binding
+// the value of 'this' is set as a window/console object. Use the following code the explore the Global element of 'this'
 
-// Principle 2
+function happyDays(feeling){
+  console.log(this)
+  return feeling;
+}
+happyDays(`fabulous`);
 
+// Principle 2 - is Implicit Binding and it states that when ever a function is called by a preceding dot that object before the dot is 'this'. Depending on the object being passed in there will be a different context for 'this' when we logout the 'this'.
 // code example for Implicit Binding
+const letsTalk = {
+  speak: 'The world is super!!',
+  speakYourMind: function(feeling) {
+    console.log(`${this.speak} and I feel ${feeling}`);
+    console.log(this);
+  }
+}
+letsTalk.speakYourMind('Fabulous');
 
-// Principle 3
+// Principle 3 - is New Binding and focuses on the concept of constructor functions which is a function that returns an object. It is its own God and creates potentially endless objects. But now we need to add the new Keyword.
 
 // code example for New Binding
 
-// Principle 4
+function HappyPerson(myFeelings) {
+  this.question = 'How are you feeling today? ';
+  this.myFeelings = myFeelings;
+  this.conversation = function() {
+    console.log(`${this.question} + ${this.myFeelings}`);
```

Comment:
don't need a plus here with a template literal (unless you wanted the plus to show in your string!)